spark-jobs/src/main/scala/filodb/downsampler/chunk/BatchExporter.scala

```diff
@@ -230,18 +232,21 @@ case class BatchExporter(downsamplerSettings: DownsamplerSettings, userStartTime
        |""".stripMargin
   }
 
-  def getRuleIfShouldExport(partKeyMap: collection.Map[String, String]): Option[ExportRule] = {
-    downsamplerSettings.exportColumnFilterMap.get(partKeyMap).flatMap { exportTableConfig =>
-      exportTableConfig.exportRules.takeWhile { rule =>
-        // step through rules while we still haven't matched a "block" filter
-        !rule.blockFilterGroups.exists { filterGroup =>
-          matchAllFilters(filterGroup, partKeyMap)
-        }
-      }.find { rule =>
-        // stop at a rule if its "allow" filters are either empty or match the partKey
-        rule.allowFilterGroups.isEmpty || rule.allowFilterGroups.exists { filterGroup =>
-          matchAllFilters(filterGroup, partKeyMap)
-        }
+  def getRuleIfShouldExport(partKeyMap: collection.Map[String, String],
+                            keyFilters: Seq[ColumnFilter],
+                            exportTableConfig: ExportTableConfig): Option[ExportRule] = {
+    if (!matchAllFilters(keyFilters, partKeyMap)) {
+      return None
+    }
+    exportTableConfig.exportRules.takeWhile { rule =>
+      // step through rules while we still haven't matched a "block" filter
+      !rule.blockFilterGroups.exists { filterGroup =>
```

Comment:
nitpick: I am guessing we already have a unit test where we have two/three matching rules and one of them is not selected for block-filters ?

Reply:
Added ✅